src/notification.cpp

```diff
@@ -52,6 +58,151 @@ NotificationManagerProxy *notificationManager()
     return notificationManagerProxyInstance();
 }
 
+QString encodeDBusCall(const QString &service, const QString &path, const QString &iface, const QString &method, const QVariantList &arguments)
+{
+    const QString space(QStringLiteral(" "));
+
+    QString s = service % space % path % space % iface % space % method;
+
+    if (!arguments.isEmpty()) {
+        QStringList args;
+        int argsLength = 0;
+
+        foreach (const QVariant &arg, arguments) {
+            // Serialize the QVariant into a Base64 encoded byte stream
+            QByteArray buffer;
+            QDataStream stream(&buffer, QIODevice::WriteOnly);
+            stream << arg;
+            args.append(space + buffer.toBase64());
+            argsLength += args.last().length();
+        }
+
+        s.reserve(s.length() + argsLength);
+        foreach (const QString &arg, args) {
+            s.append(arg);
+        }
+    }
+
+    return s;
+}
+
+QStringList encodeActions(const QHash<QString, QString> &actions)
+{
+    QStringList rv;
+
+    // Actions are encoded as a sequence of name followed by displayName
+    QHash<QString, QString>::const_iterator it = actions.constBegin(), end = actions.constEnd();
+    for ( ; it != end; ++it) {
+        rv.append(it.key());
+        rv.append(it.value());
+    }
+
+    return rv;
+}
+
+QHash<QString, QString> decodeActions(const QStringList &actions)
+{
+    QHash<QString, QString> rv;
+
+    QStringList::const_iterator it = actions.constBegin(), end = actions.constEnd();
+    for ( ; it != end; ++it) {
+        // If we have an odd number of tokens, add an empty displayName to complete the last pair
+        const QString &name(*it);
+        ++it;
+        const QString &displayName(it != end ? *it : QString());
+        rv.insert(name, displayName);
```

Comment:
This now makes it clear the the first is name and the second one is display name. Thanks :)